compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/BridgeLowering.kt

```diff
@@ -158,6 +157,48 @@ private class BridgeLowering(val context: JvmBackendContext) : ClassLoweringPass
             )
         }
 
+        // If there is an existing function that would conflict with a special bridge signature, insert the special bridge
+        // code directly as a prelude in the existing method.
+        if (!irFunction.isFakeOverride && specialOverride != null && specialOverrideSignature == ourSignature) {
+            val firstParameter = irFunction.valueParameters[0]
+            val firstParameterType = firstParameter.type
+            if (!firstParameterType.isNullable()) {
+                // Change the first parameter type to be nullable so that null checks are not generated. The checks
+                // we insert here make them superfluous.
+                irFunction.valueParameters[0] = firstParameter.copyTo(irFunction, type = firstParameterType.makeNullable())
+                // Insert an explicit null check and a return of the default value directly in the method.
+                val bodyStartOffset = irFunction.body!!.startOffset
+                val bodyEndOffset = irFunction.body!!.endOffset
+                irFunction.body = IrBlockBodyImpl(bodyStartOffset, bodyEndOffset).apply {
+                    statements += IrWhenImpl(bodyStartOffset, bodyStartOffset, context.irBuiltIns.unitType).apply {
+                        val nullConst = IrConstImpl.constNull(bodyStartOffset, bodyStartOffset, context.irBuiltIns.booleanType)
+                        val equals =
```

Comment:
Corresponding check in `createBridgeBody` is more readable -- why not copy it here, or even create an auxiliary function?

Reply:
Thank you!!! Yes, that is more readable (and writable). Changed.